nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -262,8 +264,9 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
         updateWorkflowInstance(instance);
         int parentActionId = insertWorkflowInstanceAction(instance, action);
         for (WorkflowInstance childTemplate : childWorkflows) {
-          WorkflowInstance childWorkflow = new WorkflowInstance.Builder(childTemplate).setParentWorkflowId(instance.id)
-              .setParentActionId(parentActionId).build();
+          Integer rootWorkflowId = instance.rootWorkflowId != null ? instance.rootWorkflowId : instance.id;
```

Comment:
prefer == instead of !=: instance.rootWorkflowId == null ? instance.id : instance.rootWorkflowId